polyfill/lib/intl.mjs

```diff
@@ -275,9 +283,17 @@ function formatRange(a, b) {
 }
 
 function formatRangeToParts(a, b) {
+  if (a === undefined || b === undefined) {
+    throw new TypeErrorCtor('Intl.DateTimeFormat.formatRange requires two values');
+  }
+  a = toDateTimeFormattable(a);
+  b = toDateTimeFormattable(b);
   let formatArgs = [a, b];
   let formatter;
-  if (isTemporalObject(a) || isTemporalObject(b)) {
+  if (isTemporalObject(a) !== isTemporalObject(b)) {
```

Comment:
Possible future refactor: `isTemporalObject` could return the type of object rather than a boolean